### PR TITLE
Issue #1559 Adding a commit message validation target

### PR DESCRIPTION
**Pull request by hferentschik**
- Integrating validate_commits into Circle CI

Issue #1559

### CODE REVIEW

**Comment by hferentschik** [comment]
retest this please

File: Makefile [hunk 1]
```diff
@@ -199,3 +202,7 @@ fmt:
 .PHONY: fmtcheck
 fmtcheck:
 	@gofmt -l -s $(SOURCE_DIRS) | grep ".*\.go"; if [ "$$?" = "0" ]; then exit 1; fi
+
+.PHONY: validate_commits
+validate_commits: $(GOPATH)/bin/git-validation
+	git-validation -q -run short-subject,message_regexp='(Issue #[0-9]+ [A-Z].*|cut v[0-9]+.[0-9]+.[0-9]+)' -range master..HEAD
```
Comment (gbraad) [inline Makefile]: Do we really need to be so pedantic to disallow a commit when the message starts with a lower case ? Come on...

**Comment by gbraad** [comment]
I see the same happening for the titles and I wonder how this helps:

@hferentschik changed the title from Issue #1397 preflight check for iso url to Issue #1397 Preflight check for iso url 3 days ago

Ref: https://github.com/minishift/minishift/pull/1543

Note: if you want to change the title, please make URL uppercase too...

**Comment by LalatenduMohanty** [comment]
> The target should work fine for local development and for my part I am happy if at least one CI system creates an error if the commit message is not correct.

I think it is ok as this not something we should spend a lot of time.

**Comment by LalatenduMohanty** [comment]
> Please update relevant documentation, such as https://github.com/minishift/minishift/blob/master/CONTRIBUTING.adoc to reflect this need

I think we should add `CONTRIBUTING.adoc` to Minishift documentation also. At this point of time it is only exists in the github repo. So when a contributor goes through documentation i.e. https://docs.openshift.org/latest/minishift/contributing/index.html does not see this.

File: Makefile (hunk 1 above)
Comment (hferentschik) [reply Makefile]: Can be changed ;-)

Comment (gbraad) [reply Makefile]: I see we need the option to specify like: "Issue #xxx #xxx", so "Issue #[0-9]+" would be enough I guess.

**Comment by hferentschik** [comment]
> Note: if you want to change the title, please make URL uppercase too...

Agreed

File: docs/source/contributing/developing.adoc [hunk 2]
```diff
@@ -315,6 +315,20 @@ You can correct the formatting errors yourself or instruct the violations to be
 $ make fmt
 ----
 
+[[commit-messages]]
+=== Commit messages
+
+Each commit message messages needs meet two requirements:
```
Comment (robin-owen) [inline docs/source/contributing/developing.adoc]: Repeated "message messages", should be "message".

Also, "needs meet" is grammatically incorrect (should be "needs to meet") -- I also think that the wording is more concise and clear if we replace "needs to" with "must":

"Each commit message must meet two requirements:"

File: CONTRIBUTING.adoc [hunk 3]
```diff
@@ -13,11 +13,12 @@ Before you submit a new issue, we recommend that you search the list of issues t
...
-.  Fork the repository and create a topic branch.
-.  Refer to the link:https://docs.openshift.org/latest/minishift/contributing/developing.html[developer documentation] for guidelines and tips on how to build and test Minishift.
-.  Submit a pull request with the proposed changes.
+. Create an issue describing your proposed change to the repository.
+. The repository owners will triage and respond to your issue promptly.
+. Fork the repository and create a topic branch.
+. Refer to the link:https://docs.openshift.org/latest/minishift/contributing/developing.html[developer documentation] for guidelines and tips on how to build and test Minishift.
+. Submit a pull request with the proposed changes.
+  To make reviewing and integration as smooth as possible, please run the targets link:https://docs.openshift.org/latest/minishift/contributing/developing.html#unit-tests[`test`], https://docs.openshift.org/latest/minishift/contributing/developing.html#format-source-code[`fmtcheck`] and https://docs.openshift.org/latest/minishift/contributing/developing.html#commit-messages[`validate_commits`] prior to submitting the pull request.
```
Comment (robin-owen) [inline CONTRIBUTING.adoc]: Should we clarify that these are `make` targets, or is the context enough to convey this information?

Comment (hferentschik) [reply CONTRIBUTING.adoc]: Will clarify

File: docs/source/contributing/developing.adoc (hunk 2 above)
Comment (hferentschik) [reply docs/source/contributing/developing.adoc]: Thanks, good catch.

**Comment by hferentschik** [comment]
> Please correct the typos in developing.adoc.

done

**Comment by gbraad** [comment]
retest this please

There seems to be issues with x/crypto

**Comment by gbraad** [comment]
There seems to be issues with x/crypto

# cd .; git clone https://go.googlesource.com/crypto /Users/distiller/src/golang.org/x/crypto
Cloning into '/Users/distiller/src/golang.org/x/crypto'...

**Comment by gbraad** [comment]
retest this please

**Pull request merged**